tools/gen_interp.c

```diff
@@ -377,7 +377,7 @@ fprintf(fp,"                  ngrid%%parent_grid_ratio, ngrid%%parent_grid_ratio
                strcpy( tmpstr , pp->interpu_aux_fields ) ;
 	     } else if ( down_path & FORCE_DOWN ) {
                /* by default, add the boundary and boundary tendency fields to the arg list */
-               if ( ! p->node_kind & FOURD ) {
+               if ( (! p->node_kind) & FOURD ) {
```

Comment:
```suggestion
               if ( ! (p->node_kind & FOURD) ) {
```
or
```suggestion
               if ( (~ p->node_kind) & FOURD ) {
```
`registry.h` has `#define FOURD 8` so this conditional will never be met (`!p->node_kind)` will be 0 or 1)